DESIGN.md

```diff
@@ -0,0 +1,200 @@
+
+Possible denormalized query fields on `CustomElement`:
+ * dist tags
+ * package name
+ * author
+ * keywords
+ * publish date
+ * libraries used
+ * npm downloads
+ * collections
+
+We can also denormalize fields that we want to read (but not query) along with `CustomElement`, though this is less neccessary as once we have query results we can read a `CustomElement`'s containing `PackageVersion` document easily with one extra read. This is better than reading a `PackageVersion` for potential matches only to throw them out when they fail a criteria.
+
+#### Get all elements
+
+Useful for catalog browsing and showing a set of components from a default query
+on a page or inside an embedded catalog widget. This query may be ordered by some default ranking (popularity, quality, freshness, etc) 
+
+#### Get elements by query
+
+A filtered version of the above query, queryable on any of the fields in `CustomElement`
+
+#### Get collections
+
+_TBD_
+
+#### Get authors
+
+_TBD_
+
+### Identifying Elements
+
+In several places, like URLs and cross-references from collections, we will need a way to refer to an element.
+
+Packages are not actually organized by element, they are organized by modules. Yet it's common and easy for users to refer to an element from a package, regarless of which module the element is exported from.
+
+To be convenient, we can form element identifiers from a package-name/element-name combination. If for some reason an element name is used more than once in a package (other than re-exporting it) we can add a disambiguation token, or allow the element to be referred from by module.
+
+For instance, we might have a URL to a catalog page for `@shoelace-style/shoelace/sl-button` like:
+
+`webcomponents.org/catalog/element/@shoelace-style/shoelace/sl-button/`
+
+If there are two `sl-button`s in the `@shoelace-style/shoelace` package, we can refer to the second (determined by some heuristic) like:
+
+`webcomponents.org/catalog/element/@shoelace-style/shoelace/sl-button/2`.
+
+This case should be rare since there will usually only be one "custom element export" (the `customElements.define()` or `@customElement()` call) for an element in a package.
+
+We can also refer to elements via their full package/module path:
+
+`webcomponents.org/catalog/package/@shoelace-style/shoelace/dist/components/button/button.js#sl-button`.
+
+Package versions will be encoded into the package name:
+
+`webcomponents.org/catalog/package/@shoelace-style/shoelace@2.0.0-beta.78/...`.
+
+### Requirements for Inclusion in the catalog
+
+We would like to ensure that elements in the catalog are actual custom elements, and easy to include in projects that use a variety of build systems. We can do analysis of projects as we index them to see if they meet certain requirements to facilitiate this.
+
+* `"customElements"` `package.json` field pointing to a valid custom element manifest
+* `"type": "module"` in `package.json`
```

Comment:
Mappings to modules via `exports` now seems like a valid alternative to `"type": "module"`.